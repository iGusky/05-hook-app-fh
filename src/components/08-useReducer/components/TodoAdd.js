import React from 'react'
import { useForm } from '../../../hooks/useForm';

export const TodoAdd = ({ handleAddTodo }) => {

    const [ {description}, handleInputChange, reset ] = useForm({
        description: ''
    });

    const handleSubmit = (e) => {
        e.preventDefault();

        if( description.trim().length > 1 ){          
            const newTodo = {
                id: new Date().getTime(),
                desc: description,
                done: false
            };

            handleAddTodo( newTodo );
            reset();
        }
    };

    return (
        <>
          <h4 className = "text-center">Agregar TODO</h4>
                    <form onSubmit = { handleSubmit }>
                       <div className = "form-group">
                        <input
                                type = "text"
                                name = "description"
                                placeholder = "Aprender..."
                                autoComplete = "off"
                                className = "form-control"
                                value = { description }
                                onChange = { handleInputChange }
                            />
                            <button
                                type = "submit"
                                className = "btn btn-outline-primary btn-block mt-2"
                            >
                                Agregar
                            </button>
                       </div>
                    </form>  
        </>
    )
}
