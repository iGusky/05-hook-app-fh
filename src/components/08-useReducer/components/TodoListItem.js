import React from 'react'

export const TodoListItem = ({todo, index:i, handleDelete, handleToggle}) => {
    return (
        <>
            <li 
                key = { todo.id }
                className = "list-group-item"
                >
                <p 
                    className = {`${ todo.done && 'complete'}`}
                    onClick = { () => handleToggle(todo.id) }    
                >
                    { i }. { todo.todo }
                </p>
                <button 
                    className = "btn btn-danger"
                    onClick = { () => handleDelete(todo.id) }    
                >
                    borrar
                </button>
            </li>
        </>
    )
}
