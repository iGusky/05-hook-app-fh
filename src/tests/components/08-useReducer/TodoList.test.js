import { shallow } from "enzyme"
import { TodoList } from "../../../components/08-useReducer/components/TodoList"
import { TodoListItem } from "../../../components/08-useReducer/components/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";

describe('Pruebas en <TodoList />', () => {
    
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(<TodoList
        todos = { demoTodos }
        handleDelete = { handleDelete }
        handleToggle = { handleToggle }
        />);

    test('debe de mostrarse correctamente', () => {
        expect ( wrapper ).toMatchSnapshot();
    });
    
    test('debe tener 2 <TodoListItem />', () => {
        const items = wrapper.find('TodoListItem');
        expect( items.length ).toBe( demoTodos.length );
        expect( items.at(0).prop('handleDelete') ).toEqual( expect.any(Function) );
    })
    
    
})
