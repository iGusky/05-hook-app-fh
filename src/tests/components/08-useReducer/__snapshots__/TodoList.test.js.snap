// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h4
    className="text-center"
  >
    Todos
  </h4>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={0}
      key="1"
      todo={
        Object {
          "done": false,
          "id": 1,
          "todo": "Aprender Node",
        }
      }
    />
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={1}
      key="2"
      todo={
        Object {
          "done": true,
          "id": 2,
          "todo": "Aprender Mongo",
        }
      }
    />
  </ul>
</Fragment>
`;
